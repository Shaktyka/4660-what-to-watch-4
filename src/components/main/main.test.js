import React from 'react';

import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import NameSpace from "../../reducer/name-space";
import configureStore from 'redux-mock-store';
import {BrowserRouter} from 'react-router-dom';

import Main from './main.jsx';

const FILMS_DATA = [
  {
    bgColor: `#A6B7AC`,
    cover: `https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City
      seeking revenge against Bill the Butcher, his father's killer.`,
    director: `Martin Scorsese`,
    duration: 167,
    genre: `Crime`,
    id: 1,
    isFavorite: true,
    poster: `https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
    preview: `https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    ratingCount: 370881,
    ratingScore: 8.8,
    source: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    starring: [`Leonardo DiCaprio`, `Cameron Diaz`, `Daniel Day-Lewis`],
    title: `Gangs of new york`,
    year: 2002
  }
];

const mockStore = configureStore([]);

const user = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatar: `img/1.png`
};

const genresList = [`All genres`, `Crime`, `Sci-Fi`, `Drama`];

const store = mockStore({
  [NameSpace.DATA]: {
    films: FILMS_DATA,
    isFilmsLoading: false,
    favoritesFilms: FILMS_DATA,
    genres: genresList,
    promoFilm: FILMS_DATA[0],
    isPromoLoading: false,
    loadFilmsError: ``,
    loadPromoError: ``
  },
  [NameSpace.APP_STATE]: {
    genre: `All genres`
  },
  [NameSpace.USER]: {
    authorizationStatus: `AUTH`,
    userData: user
  }
});

describe(`Main rendering`, () => {

  it(`Main renders correctly`, () => {
    const history = {
      push: () => {}
    };

    const tree = renderer
      .create(
          <BrowserRouter>
            <Provider store={store}>
              <Main
                films={FILMS_DATA}
                isLoading={false}
                loadPromoError={``}
                history={history}
              />
            </Provider>
          </BrowserRouter>, {
            createNodeMock: () => {
              return {};
            }
          }
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

});
