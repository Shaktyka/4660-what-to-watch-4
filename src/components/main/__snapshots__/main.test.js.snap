// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main rendering Main renders correctly 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
      style={
        Object {
          "backgroundColor": "#A6B7AC",
        }
      }
    >
      <img
        alt="Gangs of new york"
        src="https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          className="user-block__avatar"
          href="/mylist"
          onClick={[Function]}
          style={
            Object {
              "display": "block",
            }
          }
        >
          <img
            alt="Oliver.conner's avatar"
            height="63"
            src="https://4.react.pages.academyimg/1.png"
            width="63"
          />
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="Gangs of new york poster"
            height="327"
            src="https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Gangs of new york
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Crime
            </span>
            <span
              className="movie-card__year"
            >
              2002
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player/1"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="14"
                viewBox="0 0 18 14"
                width="18"
              >
                <use
                  xlinkHref="#in-list"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item "
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Crime
          </a>
        </li>
        <li
          className="catalog__genres-item "
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Sci-Fi
          </a>
        </li>
        <li
          className="catalog__genres-item "
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Drama
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <a
          className="small-movie-card catalog__movies-card"
          href="/films/1"
          onClick={[Function]}
        >
          <article
            id={1}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                height="175"
                poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg"
                type="video/webm"
                width="280"
              >
                Sorry, your browser doesn't support embedded video
              </video>
            </div>
            <h3
              className="small-movie-card__title"
              style={
                Object {
                  "color": "#d9ca74",
                }
              }
            >
              Gangs of new york
            </h3>
          </article>
        </a>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
